go/consensus/tendermint/full/full.go

```diff
@@ -1575,6 +958,10 @@ func New(
 	}
 	t.submissionMgr = consensusAPI.NewSubmissionManager(t, pd, viper.GetUint64(tmcommon.CfgSubmissionMaxFee))
 
+	if err := t.lazyInit(); err != nil {
```

Comment:
Isn't this problematic? Wasn't there a reason why we used lazy initialization earlier?

Reply:
Hm, maybe I'm missing something, but I don't see much difference vs before?

Before:
- `t.initialized()` was called here, and the first thing that it did was lock and call `lazyInit`
- now we call `lazyInit` first,  and then `t.initialize()`

Possibly the lazyInit should take the lock though (to match the previous behavior) but I'm not sure if it's needed.

Or is there something else that I missed?